exercises/largest-series-product/LargestSeriesProductTest.fs

```diff
@@ -1,40 +1,84 @@
-    
-[<Theory(Skip = "Remove to run test")>]
-[<InlineData("0000", 2, 0)>]
-[<InlineData("99099", 3, 0)>]
-let ``Largest product works if all spans contain zero``(digits: string) (seriesLength: int) (expected: int) =
-    largestProduct digits seriesLength |> should equal expected
-
-    
-[<Theory(Skip = "Remove to run test")>]
-[<InlineData("", 1)>]
-[<InlineData("123", 1)>]
-let ``Largest product for empty span is 1``(digits: string) (expected: int) =
-    largestProduct digits 0 |> should equal expected
-
-    
-[<Fact(Skip = "Remove to run test")>]
-let ``Cannot slice empty string with nonzero span`` () =
-    (fun () -> largestProduct "" 1 |> ignore) |> should throw typeof<Exception>
+
+[<Fact>]
+let ``Finds the largest product if span equals length`` () =
+    let digits = "29"
+    largestProduct digits 2 |> should equal 18
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can find the largest product of 2 with numbers in order`` () =
+    let digits = "0123456789"
+    largestProduct digits 2 |> should equal 72
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can find the largest product of 2`` () =
+    let digits = "576802143"
+    largestProduct digits 2 |> should equal 48
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can find the largest product of 3 with numbers in order`` () =
+    let digits = "0123456789"
+    largestProduct digits 3 |> should equal 504
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can find the largest product of 3`` () =
+    let digits = "1027839564"
+    largestProduct digits 3 |> should equal 270
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can find the largest product of 5 with numbers in order`` () =
+    let digits = "0123456789"
+    largestProduct digits 5 |> should equal 15120
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Can get the largest product of a big number`` () =
+    let digits = "73167176531330624919225119674426574742355349194934"
+    largestProduct digits 6 |> should equal 23520
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Reports zero if the only digits are zero`` () =
+    let digits = "0000"
+    largestProduct digits 2 |> should equal 0
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Reports zero if all spans include zero`` () =
+    let digits = "99099"
+    largestProduct digits 3 |> should equal 0
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Rejects span longer than string length`` () =
+    let digits = "123"
+    largestProduct digits 4 |> should equal -1
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Reports 1 for empty string and empty product (0 span)`` () =
+    let digits = ""
+    largestProduct digits 0 |> should equal 1
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Reports 1 for nonempty string and empty product (0 span)`` () =
+    let digits = "123"
+    largestProduct digits 0 |> should equal 1
+
+[<Fact(Skip = "Remove to run test")>]
+let ``Rejects empty string and nonzero span`` () =
+    let digits = ""
+    largestProduct digits 1 |> should equal -1
```

Comment:
I would like to encode an error as `None`.

Reply:
Made changes to test use `int option` as return type and use `None` for error case.